### PR TITLE
[CORE-1722] Add pachctl lifecycle tests

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

File: src/server/pps/cmds/defaults_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+			"input": {
+				"pfs": {
+					"project": "default",
+					"repo": "{{.RepoName | js}}",
+					"glob": "/*",
+					"name": "in"
+				}
+			},
+			"resource_requests": {
+				"cpu": null,
```
Comment (acohen4) [inline src/server/pps/cmds/defaults_test.go:150]: why do we set cpu to null?

**Review by acohen4 — APPROVED** [review]
Couple behavior Qs. Overall LGTM

File: src/server/pps/cmds/defaults_test.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+				"disk": "187Mi"
+			},
+			"autoscaling": false
+		}
+		EOF
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .details.resource_requests.disk | match 187Mi
+		pachctl inspect pipeline {{.PipelineName}} --raw | jq -r .details.resource_requests.memory | match 64M
```
[the remark this answers is not in the document]
Comment (robert-uhl) [reply src/server/pps/cmds/defaults_test.go:49]: I believe it is a result of https://github.com/pachyderm/pachyderm/blob/master/src/internal/minikubetestenv/deploy.go#L205

File: src/server/pps/cmds/defaults_test.go (hunk 1 above)
Comment (robert-uhl) [reply src/server/pps/cmds/defaults_test.go:150]: Copy-pasted from the API lifecycle tests. But I’ll add an explicit check for the expected results.

**Pull request merged**